tests/integration/ie-tests.sh

```diff
@@ -174,6 +174,157 @@ cleanup_user_failure() {
+  echo -e "Spark Pi Job Output: \n ${pi}"
+
+  if [ "${pi}" != "3.1" ]; then
+      exit 1
+  fi
+
+}
+
+test_example_job_in_pod() {
+  run_example_job_in_pod tests spark
+}
+
+
+run_spark_shell_in_pod() {
+  echo "run_spark_shell_in_pod ${1} ${2}"
+
+  NAMESPACE=$1
+  USERNAME=$2
+
+  echo "import scala.math.random" > test-spark-shell.scala
+  echo "val slices = 10" >> test-spark-shell.scala
+  echo "val n = math.min(100000L * slices, Int.MaxValue).toInt" >> test-spark-shell.scala
+  echo "val count = spark.sparkContext.parallelize(1 until n, slices).map { i => val x = random * 2 - 1; val y = random * 2 - 1;  if (x*x + y*y <= 1) 1 else 0;}.reduce(_ + _)" >> test-spark-shell.scala
+  echo "println(s\"Pi is roughly \${4.0 * count / (n - 1)}\")" >> test-spark-shell.scala
+  echo "System.exit(0)" >> test-spark-shell.scala
+
+  SPARK_SHELL_COMMANDS=$(cat ./test-spark-shell.scala)
+
+  echo -e "$(kubectl exec testpod -- env UU="$USERNAME" NN="$NAMESPACE" CMDS="$SPARK_SHELL_COMMANDS" /bin/bash -c 'echo "$CMDS" | spark-client.spark-shell --username $UU --namespace $NN')" > spark-shell.out
+
+  pi=$(cat spark-shell.out  | grep "^Pi is roughly" | rev | cut -d' ' -f1 | rev | cut -c 1-3)
+  echo -e "Spark-shell Pi Job Output: \n ${pi}"
+  rm spark-shell.out test-spark-shell.scala
+  if [ "${pi}" != "3.1" ]; then
+      exit 1
+  fi
+}
+
+test_spark_shell_in_pod() {
+  run_spark_shell_in_pod tests spark
+}
+
+run_pyspark_in_pod() {
+  echo "run_pyspark_in_pod ${1} ${2}"
+
+  NAMESPACE=$1
+  USERNAME=$2
+
+  echo "import sys" > test-pyspark.py
+  echo "from random import random" >> test-pyspark.py
+  echo "from operator import add" >> test-pyspark.py
+  echo "from pyspark.context import SparkContext" >> test-pyspark.py
+  echo "from pyspark.sql.session import SparkSession" >> test-pyspark.py
+  echo "sc = SparkContext()" >> test-pyspark.py
+  echo "spark = SparkSession(sc)" >> test-pyspark.py
+  echo "for conf in spark.sparkContext.getConf().getAll(): print (conf)" >> test-pyspark.py
+  echo "partitions = 10" >> test-pyspark.py
+  echo "n = 1000000 * partitions" >> test-pyspark.py
+  echo "def f(_: int) -> float:" >> test-pyspark.py
+  echo "     x, y = random(), random()" >> test-pyspark.py
+  echo "     return x * x + y * y < 1 " >> test-pyspark.py
+  echo "count = spark.sparkContext.parallelize(range(n), partitions).filter(f).count()" >> test-pyspark.py
+  echo "print (\"Pi is roughly %f\" % (4.0 * count / n))" >> test-pyspark.py
+
+  PYSPARK_COMMANDS=$(cat ./test-pyspark.py)
+
+  echo -e "$(kubectl exec testpod -- env UU="$USERNAME" NN="$NAMESPACE" CMDS="$PYSPARK_COMMANDS" /bin/bash -c 'echo "$CMDS" | spark-client.pyspark --username $UU --namespace $NN')" > pyspark.out
+
+  cat pyspark.out
+  pi=$(cat pyspark.out  | grep "Pi is roughly" | tail -n 1 | rev | cut -d' ' -f1 | rev | cut -c 1-3)
+  echo -e "Pyspark Pi Job Output: \n ${pi}"
+  rm test-pyspark.py pyspark.out
+  if [ "${pi}" != "3.1" ]; then
+      exit 1
```

Comment:
Maybe some error message for aborting the test?
Newbies do not know the magic value `3.1` :-)